lit_gpt/model.py

```diff
@@ -320,20 +318,15 @@ def build_rope_cache(
     https://github.com/labmlai/annotated_deep_learning_paper_implementations/blob/master/license.
     """
     # $\Theta = {\theta_i = 10000^{\frac{2(i-1)}{d}}, i \in [1, 2, ..., \frac{d}{2}]}$
-    theta = 1.0 / (base ** (torch.arange(0, n_elem, 2, device=device) / n_elem))
+    theta = 1.0 / (base ** (torch.arange(0, n_elem, 2, device=device).float() / n_elem))
 
     # Create position indexes `[0, 1, ..., seq_len - 1]`
     seq_idx = torch.arange(seq_len, device=device) / condense_ratio
 
     # Calculate the product of position index and $\theta_i$
     idx_theta = torch.outer(seq_idx, theta).repeat(1, 2)
 
-    cos, sin = torch.cos(idx_theta), torch.sin(idx_theta)
-
-    # this is to mimic the behaviour of complex32, else we will get different results
-    if dtype in (torch.float16, torch.bfloat16, torch.int8):
-        return cos.half(), sin.half()
-    return cos, sin
```

Comment:
What are the implications? These changes were necessary to have parity with the original llama rope cache implementation.

Reply:
Not doing this matches the HF Rope implementations. HF keeps these in float32.

This piece of code comes from lit-llama, where it was ported from the original facebookresearch repo. Since this repo uses the HF implementations as references instead, I think it's fine to remove this.

The mistral original release still uses complex numbers: https://github.com/mistralai/mistral-src/blob/main/mistral/rope.py

I don't think there's a solution that is numerically precise and performant across implementations. We just need to choose which reference implementation we compare this with.